src/features/selectedPost.ts

```diff
@@ -0,0 +1,27 @@
+import { PayloadAction, createSlice } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+
+export interface SelectedPostState {
+  selectedPost: Post | null;
+}
+
+const initialState: SelectedPostState = {
+  selectedPost: null,
+};
+
+export const selectedPostSlice = createSlice({
+  name: 'selectedPost',
+  initialState,
+  reducers: {
+    currentPost: (state, action: PayloadAction<Post | null>) => {
```

Comment:
since it's a function let's rename it accordingly (add verb), for example: `setCurrentPost` or `selectPost`, etc